csrc/ir/utils.cpp

```diff
@@ -1414,8 +1411,10 @@ bool matchMatmulPatterns(const UnaryOp* cast_op, MatmulInputs* matmul_inp) {
 
   // Verify assumptions for MmaOp hold. Assign the values to Mma operands.
   MmaOpUtils::verifyMmaOpForEvaluation(mma, final_out_dtype);
-  matmul_inp->mma_lhs = mma->inA();
-  matmul_inp->mma_rhs = mma->inB();
+  // Get the non-broadcasted values to avoid inferring squeeze dimensions.
+  matmul_inp->mma_lhs = mma->inA()->definition()->input(0);
+  matmul_inp->mma_rhs = mma->inB()->definition()->input(0);
+  matmul_inp->mma_layout = mma->layout().value();
```

Comment:
This might better be done with `mma_utils::getMmaLayout(fusion);` than looking at what's stored in the MmaOp.

Reply:
So this function also returns `MmaLayout`. Are we using MmaLayout for two different definitions?